crates/iceberg/src/error.rs

```diff
@@ -333,6 +335,28 @@ define_from_err!(
 
 define_from_err!(std::io::Error, ErrorKind::Unexpected, "IO Operation failed");
 
+/// Converts a timestamp in milliseconds to `DateTime<Utc>`, handling errors.
+///
+/// # Arguments
+///
+/// * `timestamp_ms` - The timestamp in milliseconds to convert.
+///
+/// # Returns
+///
+/// This function returns a `Result<DateTime<Utc>, Error>` which is `Ok` with the `DateTime<Utc>` if the conversion is successful,
+/// or an `Err` with an appropriate error if the timestamp is ambiguous or invalid.
+pub(crate) fn timestamp_ms_to_utc(timestamp_ms: i64) -> Result<DateTime<Utc>> {
+    match Utc.timestamp_millis_opt(timestamp_ms) {
+        chrono::LocalResult::Single(t) => Ok(t),
+        chrono::LocalResult::Ambiguous(_, _) => Err(Error::new(
```

Comment:
How about print the detailed info in error?

Reply:
I am not quite sure what do you mean with detailed info in error - we already have the context "timestamp value". Would you add a  more verbose message?

Reply:
> I am not quite sure what do you mean with detailed info in error - we already have the context "timestamp value". Would you add a more verbose message?

I want to carry the info returned in `LocalResult::Ambiguous(_, _)`. But I'm guessing they are not useful?